src/js/components/TextInput/TextInput.js

```diff
@@ -376,43 +391,88 @@ const TextInput = forwardRef(
           }
           onKeyDown={onKeyDown}
         >
-          <StyledTextInput
-            ref={ref || inputRef}
-            id={id}
-            name={name}
-            autoComplete="off"
-            plain={plain}
-            placeholder={
-              typeof placeholder === 'string' ? placeholder : undefined
-            }
-            icon={icon}
-            reverse={reverse}
-            focus={focus}
-            {...rest}
-            defaultValue={renderLabel(defaultValue)}
-            value={renderLabel(value)}
-            readOnly={readOnly}
-            onFocus={event => {
-              setFocus(true);
-              if (suggestions && suggestions.length > 0) {
-                announce(messages.suggestionsExist);
-                openDrop();
+          {handleTextSelect.current ? (
```

Comment:
I don't think we should have a condition of 'handleTextSelect' to control the whole rendering of StyledTextInput. Lots of duplicated code here.